### ISSUE TITLE
Add support for "common" NX workspace

### ISSUE THREAD

**Issue by montella1507** [ngneat/scam#3]
If you generate NX workspace without libs (empty) it uses Workspace.json and not Angular.json so you will end with:

`The generate command requires to be run in an Angular project, but a project definition could not be found.`

"nx g @ngneat/scam:component name"
ends with:

```
FileDoesNotExistException [Error]: Path "/D:/apps/assist-mobile/src/app/name/name.component.ts" does not exist.
    at HostTree.overwrite (D:\GIT\Cue\frontends\node_modules\@angular-devkit\schematics\src\tree\host-tree.js:250:19)
    at D:\GIT\Cue\frontends\node_modules\@ngneat\scam\src\utils.js:57:14
    at MergeMapSubscriber.project (D:\GIT\Cue\frontends\node_modules\@ngneat\scam\node_modules\@angular-devkit\schematics\src\rules\call.js:75:24)
    at MergeMapSubscriber._tryNext (D:\GIT\Cue\frontends\node_modules\@ngneat\scam\node_modules\rxjs\internal\operators\mergeMap.js:67:27)
    at MergeMapSubscriber._next (D:\GIT\Cue\frontends\node_modules\@ngneat\scam\node_modules\rxjs\internal\operators\mergeMap.js:57:18)
    at MergeMapSubscriber.Subscriber.next (D:\GIT\Cue\frontends\node_modules\@ngneat\scam\node_modules\rxjs\internal\Subscriber.js:66:18)
    at MergeMapSubscriber.notifyNext (D:\GIT\Cue\frontends\node_modules\@ngneat\scam\node_modules\rxjs\internal\operators\mergeMap.js:93:26)
    at SimpleInnerSubscriber._next (D:\GIT\Cue\frontends\node_modules\@ngneat\scam\node_modules\rxjs\internal\innerSubscribe.js:27:21)
    at SimpleInnerSubscriber.Subscriber.next (D:\GIT\Cue\frontends\node_modules\@ngneat\scam\node_modules\rxjs\internal\Subscriber.js:66:18)
    at TapSubscriber._next (D:\GIT\Cue\frontends\node_modules\@ngneat\scam\node_modules\rxjs\internal\operators\tap.js:65:26)
Path "/D:/apps/assist-mobile/src/app/name/name.component.ts" does not exist.
```

**Comment by NetanelBasal** [comment]
Do you want to create a PR?

**Comment by montella1507** [comment]
Good question, may be good way how to start with schematics.

**Comment by KingDarBoja** [comment]
Maybe these resources can be helpful to start with schematics:
- https://angular.io/guide/schematics-authoring
- https://auth0.com/blog/create-custom-schematics-with-nx/#Creating-a-Custom-Auth-Schematic

**Comment by LayZeeDK** [comment]
@montella1507 please elaborate on the steps to reproduce.

**Comment by montella1507** [comment]
1)  npx create-nx-workspace@latest ngneat
      npx: installed 48 in 2.654s
2)  What to create in the new workspace · empty
3)  Use Nx Cloud? (It's free and doesn't require registration.) · No
4) cd ngneat
5) npm install --save-dev @nrwl/angular
6)   nx g @nrwl/angular:application
√ What name would you like to use for the application? · test
√ Which stylesheet format would you like to use? · scss
√ Would you like to configure routing for this application? (y/N) · false
- Installing packages (npm)...
7) npm i @ngneat/scam --save-dev
8a) ng g @ngneat/scam:component name
**The generate command requires to be run in an Angular project, but a project definition could not be found.**

8b) nx g @ngneat/scam:component name
```
FileDoesNotExistException [Error]: Path "/d:/apps/test/src/app/name/name.component.ts" does not exist.
    at HostTree.overwrite (d:\GIT\ngneat\node_modules\@angular-devkit\schematics\src\tree\host-tree.js:250:19)
    at d:\GIT\ngneat\node_modules\@ngneat\scam\src\utils.js:54:14
    at MergeMapSubscriber.project (d:\GIT\ngneat\node_modules\@angular-devkit\schematics\src\rules\call.js:75:24)
    at MergeMapSubscriber._tryNext (d:\GIT\ngneat\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:67:27)
    at MergeMapSubscriber._next (d:\GIT\ngneat\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:57:18)
    at MergeMapSubscriber.Subscriber.next (d:\GIT\ngneat\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\Subscriber.js:66:18)
    at MergeMapSubscriber.notifyNext (d:\GIT\ngneat\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:93:26)
    at SimpleInnerSubscriber._next (d:\GIT\ngneat\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\innerSubscribe.js:27:21)
    at SimpleInnerSubscriber.Subscriber.next (d:\GIT\ngneat\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\Subscriber.js:66:18)
    at TapSubscriber._next (d:\GIT\ngneat\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\tap.js:65:26)
```


Angular CLI: 11.2.5
Node: 14.15.1
OS: win32 x64

Angular: 11.2.12
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: <error>


Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1102.11
@angular-devkit/build-angular   0.1102.11
@angular-devkit/core            11.2.11
@angular-devkit/schematics      11.2.11
@schematics/angular             11.2.11
@schematics/update              0.1102.5 (cli-only)
rxjs                            6.6.7
typescript                      4.1.5

minimal repro:
[ngneat.zip](https://github.com/ngneat/scam/files/6419807/ngneat.zip)

**Comment by LayZeeDK** [comment]
Ah, there are even issues with the `--project` options.

Errors with different options on Windows 10:
```
PS C:\projects\sandbox\ngneat-scam-issue3> nx generate @ngneat/scam:component scam --project=scam-app
FileDoesNotExistException [Error]: Path "/C:/apps/scam-app/src/app/scam/scam.component.ts" does not exist.
    at HostTree.overwrite (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\src\tree\host-tree.js:250:19)
    at C:\projects\sandbox\ngneat-scam-issue3\node_modules\@ngneat\scam\src\utils.js:54:14
    at MergeMapSubscriber.project (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\src\rules\call.js:75:24)       
    at MergeMapSubscriber._tryNext (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:67:27)
    at MergeMapSubscriber._next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:57:18)
    at MergeMapSubscriber.Subscriber.next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\Subscriber.js:66:18)
    at MergeMapSubscriber.notifyNext (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:93:26)
    at SimpleInnerSubscriber._next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\innerSubscribe.js:27:21)
    at SimpleInnerSubscriber.Subscriber.next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\Subscriber.js:66:18)
    at TapSubscriber._next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\tap.js:65:26)
Path "/C:/apps/scam-app/src/app/scam/scam.component.ts" does not exist.
PS C:\projects\sandbox\ngneat-scam-issue3> nx generate @ngneat/scam:component scam --project=scam-app --path=scam
FileDoesNotExistException [Error]: Path "/C:/projects/sandbox/ngneat-scam-issue3/scam/scam/scam.component.ts" does not exist.
    at HostTree.overwrite (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\src\tree\host-tree.js:250:19)
    at C:\projects\sandbox\ngneat-scam-issue3\node_modules\@ngneat\scam\src\utils.js:54:14
    at MergeMapSubscriber.project (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\src\rules\call.js:75:24)       
    at MergeMapSubscriber._tryNext (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:67:27)
    at MergeMapSubscriber._next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:57:18)
    at MergeMapSubscriber.Subscriber.next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\Subscriber.js:66:18)
    at MergeMapSubscriber.notifyNext (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:93:26)
    at SimpleInnerSubscriber._next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\innerSubscribe.js:27:21)
    at SimpleInnerSubscriber.Subscriber.next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\Subscriber.js:66:18)
    at TapSubscriber._next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\tap.js:65:26)
Path "/C:/projects/sandbox/ngneat-scam-issue3/scam/scam/scam.component.ts" does not exist.
PS C:\projects\sandbox\ngneat-scam-issue3> nx generate @ngneat/scam:component scam --project=scam-app --path=src/app
FileDoesNotExistException [Error]: Path "/C:/projects/sandbox/ngneat-scam-issue3/src/app/scam/scam.component.ts" does not exist.
    at HostTree.overwrite (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\src\tree\host-tree.js:250:19)
    at C:\projects\sandbox\ngneat-scam-issue3\node_modules\@ngneat\scam\src\utils.js:54:14
    at MergeMapSubscriber.project (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\src\rules\call.js:75:24)       
    at MergeMapSubscriber._tryNext (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:67:27)
    at MergeMapSubscriber._next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:57:18)
    at MergeMapSubscriber.Subscriber.next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\Subscriber.js:66:18)
    at MergeMapSubscriber.notifyNext (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:93:26)
    at SimpleInnerSubscriber._next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\innerSubscribe.js:27:21)
    at SimpleInnerSubscriber.Subscriber.next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\Subscriber.js:66:18)
    at TapSubscriber._next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\tap.js:65:26)
Path "/C:/projects/sandbox/ngneat-scam-issue3/src/app/scam/scam.component.ts" does not exist.
PS C:\projects\sandbox\ngneat-scam-issue3> nx generate @ngneat/scam:component scam --project=scam-app --path=apps/scam-app/src/app
FileDoesNotExistException [Error]: Path "/C:/projects/sandbox/ngneat-scam-issue3/apps/scam-app/src/app/scam/scam.component.ts" does not exist.   
    at HostTree.overwrite (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\src\tree\host-tree.js:250:19)
    at C:\projects\sandbox\ngneat-scam-issue3\node_modules\@ngneat\scam\src\utils.js:54:14
    at MergeMapSubscriber.project (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\src\rules\call.js:75:24)       
    at MergeMapSubscriber._tryNext (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:67:27)
    at MergeMapSubscriber._next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:57:18)
    at MergeMapSubscriber.Subscriber.next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\Subscriber.js:66:18)
    at MergeMapSubscriber.notifyNext (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\mergeMap.js:93:26)
    at SimpleInnerSubscriber._next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\innerSubscribe.js:27:21)
    at SimpleInnerSubscriber.Subscriber.next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\Subscriber.js:66:18)
    at TapSubscriber._next (C:\projects\sandbox\ngneat-scam-issue3\node_modules\@angular-devkit\schematics\node_modules\rxjs\internal\operators\tap.js:65:26)
Path "/C:/projects/sandbox/ngneat-scam-issue3/apps/scam-app/src/app/scam/scam.component.ts" does not exist.
```

**Comment by LayZeeDK** [comment]
Same issue with Angular CLI:
```
PS C:\projects\sandbox\ngx-ngneat-scam> ng generate @ngneat/scam:component scam
Option "entryComponent" is deprecated: Since version 9.0.0 with Ivy, entryComponents is no longer necessary.
Path "/C:/projects/scam-app/src/app/scam/scam.component.ts" does not exist.
```

**Comment by phhien203** [comment]
Hi guys, I fixed the error "Path ... does not exist" in this PR https://github.com/ngneat/scam/pull/9. I saw that the schematic supports Nx workspace created by using `--preset=angular`. In case Angular workspace created by calling `nx g @nrwl/angular:application`, the error still happens.

**Comment by NetanelBasal** [comment]
Fixed in v2.1.0
